shared/utils/codeql/utils/FileSystem.qll

```diff
@@ -0,0 +1,214 @@
+/**
+ * A shared library for reasoning about files and folders.
+ * This file contains 3 modules:
+ * - `BasicFileSig` is a basic interface that every language is expected to implement.
```

Comment:
I don't see the definition of `BasicFileSig` anywhere, does this comment need updating?

Reply:
Yes it does.  
When I wrote that comment I had 3 modules as described, but I removed the "full implementation as a signature" module.